application/src/main/java/org/opentripplanner/inspector/vector/geofencing/GeofencingZonesLayerBuilder.java

```diff
@@ -29,15 +31,36 @@ public GeofencingZonesLayerBuilder(Graph graph, LayerParameters layerParameters)
 
   @Override
   protected List<Geometry> getGeometries(Envelope query) {
-    return graph
+    Map<String, GeofencingZone> uniqueZones = new HashMap<>();
+
+    graph
       .findVertices(query)
       .stream()
-      .filter(se -> !(se.rentalRestrictions() instanceof NoRestriction))
-      .map(vertex -> {
-        Geometry geometry = GeometryUtils.getGeometryFactory().createPoint(vertex.getCoordinate());
-        geometry.setUserData(vertex);
+      .filter(v -> !(v.rentalRestrictions() instanceof NoRestriction))
+      .forEach(vertex -> {
+        extractGeofencingZones(vertex.rentalRestrictions(), uniqueZones);
```

Comment:
I think it might be better to use some stream collect to build the map instead of doing this kind of a side effect inside the stream. Although, it probably doesn't really matter.

Reply:
I agree with both of you. The intermediate Map wasn't needed anyway. Sloppy of me - I cleaned it up a bit.